### DISCUSSION TITLE
Implicit Coreference Resolution

### DISCUSSION THREAD

**Discussion by ChenKua** [Aggregate-Intellect/sherpa#324] [Ideas]
**Description**:

- Provide a clear and detailed explanation of the problem you're encountering.
- Include any relevant context or background information.
- Mention the expected behavior if applicable.
- Share any error messages or logs you've encountered.

Our slack bot might be unable to identify implicit coreference in certain cases. For example, during the previous conversation, we asked questions on vector store in langchain, then when we asked for a Python example. Sherpa did not get we want a Python example (of vector store in langchain)

**Steps to Reproduce:**

- Provide a step-by-step guide on how to reproduce the issue.
- Include specific details such as input data, configurations, or any necessary prerequisites.
- If applicable, mention the browser, operating system, or environment you're using.

```
_USER: @sherpa how can I use vector store in langchain
sherpa: blablabla
USER: @sherpa can you give me a python example?
sherpa: here is a python example of calculating the factorial of a number. bla bla bla
```

The example has nothing to do with vector store or langchain

**Actual Results:**

- Describe the actual outcome or behavior you observed.
- Include any error messages, warnings, or relevant logs.

See above

**Expected Results:**

- Clearly state what you expected to happen instead.
- This helps developers understand the desired behavior.
I would expect the bot to understand the implicit coreference and give a response with correct information.

**Additional Information:**

- Provide any additional details that may be helpful but were not covered above.
- Include screenshots, code snippets, or links to relevant resources if applicable.

N/A

**Reproducibility**:

- Mention whether the issue is reproducible consistently or intermittently.
- Include any relevant information on the frequency or conditions under which it occurs.
- The actual answer might vary but the general pattern is reproducible.

In certain cases, the bot is able to catch the implicit coreference.
```
USER: Do you know vector database
CHATGPT: A vector database is a type of database  blablabl
USER: Can you give a python code example
CHATGPT: Here's a simple Python code example that demonstrates how to work with a vector database using the Faiss library blabla
```

In the following cases, our bot failed as well
In certain cases, the bot is able to catch the implicit coreference.
```
USER: what is agent in langchain
sherpa: an agent in LangChain is blablabl
USER: Can you give a python code example
sherpa: Here's a simple Python code example that demonstrates how to calculate the factorial of a number
```
<img width="591" alt="image" src="https://github.com/Aggregate-Intellect/practical-llms/assets/58306103/093197f1-b090-4bcf-b3c7-49fd3c1f2a31">

**Possible Solutions:**

- If you have any ideas or suggestions for how the issue could be resolved, mention them here.
- This can help guide the discussion and provide potential starting points for developers.

Related Issues:

> If there are any related GitHub issues or pull requests, reference them here.
> This helps create connections and provides additional context.
N/A

**Steps Taken So Far:**

- Outline any steps you've already taken to investigate or resolve the issue.
- This helps developers understand what has been attempted and avoids duplication of effort.

N/A

**Environment**:

- Provide details about the environment you're working in (e.g., programming language, versions, libraries, frameworks).
- Mention any relevant dependencies or configurations.

N/A

**Comment by ChenKua** [comment]
Have not push code to remote because a 10GB LLM is used and I dont think I can push such a large file to github. Before we can figure out how to serve Llama 2 in a proper way, I don't think the code is ready to be merged.

**Comment by amirfz** [comment]
this is currently blocked by issue #81
